### ISSUE TITLE
Import of numpy fails

### ISSUE THREAD

**Issue by gehlhaarPfizer** [pyinstaller/pyinstaller#4058]
PyInstaller 3.4, numpy 1.16, Windows 10, Python 3.6.8

I am facing a critical problem whereby import of numpy in my script fails.

A test script of "import numpy" run with "pyinstaller.exe --debug --clean test.py" generates a program that fails:

```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\__init__.py", line 151, in <module>
    from . import ctypeslib
  File "<frozen importlib._bootstrap>", line 1023, in _handle_fromlist
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\ctypeslib.py", line 369, in <module>
    _typecodes = _get_typecodes()
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\ctypeslib.py", line 358, in _get_typecodes
    return {_dtype(ctype).str: ctype for ctype in simple_types}
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\ctypeslib.py", line 358, in <dictcomp>
    return {_dtype(ctype).str: ctype for ctype in simple_types}
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'numpy.core._dtype_ctypes'
```

This works interactively with Python, and also works if I replace the 'numpy' directory generated by PyInstaller with the one in my python/Lib/site-packages directory.

In my "real" script I get a different error, loading the multiarray DLL:

```
Traceback (most recent call last):
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\core\__init__.py", line 16, in <module>
    from . import multiarray
  File "<frozen importlib._bootstrap>", line 1023, in _handle_fromlist
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\core\multiarray.py", line 12, in <module>
    from . import overrides
  File "<frozen importlib._bootstrap>", line 1023, in _handle_fromlist
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\home\gehlhaar\svn\all\3rdparty\Windows_cl\python\Lib\site-packages\numpy\core\overrides.py", line 9, in <module>
    from numpy.core._multiarray_umath import add_docstring, ndarray
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 658, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 571, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 922, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.
```

It is fixed with the same deep copy. 

This is very unsettling -- I would have expected numpy to work fine out of the box.

**Comment by gehlhaarPfizer** [comment]
Found error in my script -- the simple one still fails, although the more complex one works.

**Comment by bjones1** [comment]
Also, this is fixed in numpy 1.6.1.
